ThirdParty/UIImageColors.swift

```diff
@@ -275,4 +275,18 @@ extension UIImage {
 
         return result
     }
+    
+    var averageColor: UIColor? {
+        guard let inputImage = CIImage(image: self) else { return nil }
+        let extentVector = CIVector(x: inputImage.extent.origin.x, y: inputImage.extent.origin.y, z: inputImage.extent.size.width, w: inputImage.extent.size.height)
+
+        guard let filter = CIFilter(name: "CIAreaAverage", parameters: [kCIInputImageKey: inputImage, kCIInputExtentKey: extentVector]) else { return nil }
+        guard let outputImage = filter.outputImage else { return nil }
+
+        var bitmap = [UInt8](repeating: 0, count: 4)
+        let context = CIContext(options: [.workingColorSpace: kCFNull])
+        context.render(outputImage, toBitmap: &bitmap, rowBytes: 4, bounds: CGRect(x: 0, y: 0, width: 1, height: 1), format: .RGBA8, colorSpace: nil)
+
+        return UIColor(red: CGFloat(bitmap[0]) / 255, green: CGFloat(bitmap[1]) / 255, blue: CGFloat(bitmap[2]) / 255, alpha: CGFloat(bitmap[3]) / 255)
+    }
```

Comment:
Oof! 😅  Two comments here:

1) This is doing a lot for a calculated variable. Would it make more sense for it to be a function?
2) I've never worked with some of this stuff and have no idea what the point is to what's happening in a few places. Can we add a few comments here and there as to what's happening? Or a documentation comment as to what's happening for the whole thing? If I need to debug this at some point, some direction/help would be nice. :)

Reply:
Ah, for sure let me put a comment :)